File: westcoast/src/main/java/frc/commands/DistanceMotor.java
package frc.commands;

import edu.wpi.first.wpilibj.SpeedController;
import edu.wpi.first.wpilibj2.command.CommandBase;

public class DistanceMotor extends CommandBase {
  private final SpeedController motor;
  private int i;
  private boolean end;
  private final 
  double time;
  private double speed;

  public DistanceMotor(
    SpeedController motor,
    double speed,
    double time
  ) {
    this.motor = motor;
    this.speed = speed;
    this.time = time*50;
  }

  @Override
  public void initialize() {
    i = 0;
    end = false;
  }

  @Override
  public void execute() {
    if(i <= time){
      motor.set(speed);
      i=i+1;
    }
    else{
      end = true;
    }
    System.out.println(i);
  }

  @Override
  public void end(boolean interrupted) {
    motor.stopMotor();
    i = 0;
   }

  @Override
  public boolean isFinished() {
    return end;
  }
}

File: westcoast/src/main/java/frc/commands/motors/DistanceMotor.kt
package frc.commands.motors

import edu.wpi.first.wpilibj.SpeedController
import edu.wpi.first.wpilibj2.command.CommandBase

class DistanceMotor(
    private val motor: SpeedController,
    private val speed: Double,
    time: Double
) : CommandBase() {
  private var i = 0
  private var end = false
  private val time: Double
  override fun initialize() {
    i = 0
    end = false
  }

  override fun execute() {
    if (i <= time) {
      motor.set(speed)
      i = i + 1
    }
    else {
      end = true
    }
    println(i)
  }

  override fun end(interrupted: Boolean) {
    motor.stopMotor()
    i = 0
  }

  override fun isFinished(): Boolean {
    return end
  }

  init {
    this.time = time * 50
  }
}